### ISSUE TITLE
All Travis and Appveyor builds failing

### ISSUE THREAD

**Issue by timtrice** [ropensci/rrricanes#121]
Currently, testing rricanes on R 3.3.3 thru devel, linux and osx.  Appveyoron R 3.3 to 3.4 and devel (so no 3.5?). 

All builds fail. 

## Travis

As of now, [these are the issues](https://travis-ci.org/ropensci/rrricanes/builds/472850552):

* Linux builds 497.1,  497.3, 497.5, 497.7, 497.9:

```
configure: GDAL: 1.10.1
checking GDAL version >= 1.11.4... no
configure: error: upgrade GDAL to 1.11.4 or later
ERROR: configuration failed for package ‘rgdal’
* removing ‘/home/travis/R/Library/rgdal’
Error in i.p(...) : 
  (converted from warning) installation of package ‘rgdal’ had non-zero exit status
```

* OSX builds 497.2:

```
Warning messages:
1: In install.packages("devtools") :
  installation of package ‘git2r’ had non-zero exit status
2: In install.packages("devtools") :
  installation of package ‘usethis’ had non-zero exit status
3: In install.packages("devtools") :
  installation of package ‘devtools’ had non-zero exit status
Installing R packages from GitHub: timtrice/rrricanesdata
0.15s$ Rscript -e 'devtools::install_github(c("timtrice/rrricanesdata"), build_vignettes = FALSE)'
Error in loadNamespace(name) : there is no package called ‘devtools’
```

* OSX builds 497.4, 497.6, 497.8, 497.10, 497.12, 497.14:

```
Error: (converted from warning) unable to access index for repository https://timtrice.github.io/drat/bin/macosx/el-capitan/contrib/3.4:
  cannot open URL 'https://timtrice.github.io/drat/bin/macosx/el-capitan/contrib/3.4/PACKAGES'
Execution halted
```

* OSX builds 497.11, 497.13:

```
Quitting from lines 251-256 (gis_data.Rmd) 
Error: processing vignette 'gis_data.Rmd' failed with diagnostics:
No tidy method for objects of class SpatialPointsDataFrame
```

## Appveyor

Because Appveyor doesn't run subsequent tests if one fails, I only have one error as the first build fails.

[Current issue](https://ci.appveyor.com/project/timtrice/rrricanes-g4dos/builds/21261663):

```
Error: (converted from warning) unable to access index for repository https://timtrice.github.io/drat/bin/windows/contrib/3.4:
  cannot open URL 'https://timtrice.github.io/drat/bin/windows/contrib/3.4/PACKAGES'
```

**Comment by timtrice** [comment]
For travis, the last passing commit was [46f20b5](https://github.com/ropensci/rrricanes/blob/46f20b50d704f22fc59a0bb083c88a3090a4664c/.travis.yml); appveyor was [4323c517](https://github.com/ropensci/rrricanes/blob/4323c517411b43dd08c1c03ab1194a74d2cf5fc9/appveyor.yml). 

git diff in travis:

```
diff --git a/.travis.yml b/.travis.yml
index 096cc23..1439b0b 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -3,28 +3,25 @@ warnings_are_errors: true
 addons:
   apt:
     packages:
-    - libgdal-dev
-    - libproj-dev
-matrix:
-  include:
-  - os: linux
-    dist: trusty
-    sudo: false
-    r: 3.3.3
-  - os: linux
-    dist: trusty
-    sudo: false
-    r: release
-  - os: linux
-    dist: trusty
-    sudo: false
-    r: devel
-  - os: osx
-    r: 3.3.3
-  - os: osx
-    r: release
+      - gdal-bin
+      - libgdal-dev
+      - libgdal1-dev
+      - libproj-dev
+dist: trusty
+os:
+  - linux
+  - osx
+r:
+  - 3.3.3
+  - 3.4.0
+  - 3.4.1
+  - 3.4.2
+  - 3.4.3
+  - release
+  - devel
 r_packages:
   - covr
+  - devtools
 r_github_packages:
   - timtrice/rrricanesdata
 cache: packages
@@ -33,6 +30,15 @@ after_success:
   - test $TRAVIS_PULL_REQUEST == "false" &&
     test $TRAVIS_BRANCH == "master" &&
     bash deploy.sh
+before_install:
+  - sudo add-apt-repository ppa:ubuntugis/ubuntugis-unstable --yes
+  - sudo apt-get --yes --force-yes update -qq
+  - sudo apt-get install -y gdal-bin libgdal-dev libgdal1-dev libproj-dev
+  - curl -OL http://raw.github.com/craigcitro/r-travis/master/scripts/travis-tool.sh
+  - chmod 755 ./travis-tool.sh
+  - ./travis-tool.sh bootstrap
+  - ./travis-tool.sh install_deps
+  - ./travis-tool.sh install_github jimhester/covr hadley/xml2
 env:
   global:
     secure: t3RJn4gGOnWDc2DSBOHwZ+DkLTZsmo2f9JDfg0H9ugzhZQFvXWy9/+HCak4t00aCpRojkMadqmQPORoeGCvzbdZV6Iu+ueFBb9JPD6CVgBiUYLcBL0aaYtPydKpw5EAFmvoHQo2J/p6bVtVvpBPOf+OVaeLqzcKFD8mxJtVl/sbSuZzCUKhs2KPIEtfpYZuchlZKRV6K+vrDomQytqQJeTlc5TTqY1VMOyY2E1eRgTxpTnN44caWSiqAmMkBx403TmFRpA0XZzsxxmUxTpXrNayT6S0qdn/VClyTTvJb/E92jFjm6c2MxCL6D5Xw7mv51srHf4qIxf3eLyj2ilZUKB4aNeecr5Y1/NeBgmx9K+WD16e9TJYOfR0J8i2HikBLgSgFCvU8D5p3sC6gI/+GCPBXyAEVcndnPUBCaRqfz4eeu3CkxiWgzWwabBoqjas/jp7gyOaj1lwEqOm+WNgA4E/7Klv161CXVHPLyvC7w/7mwVLK6PYN/6yyLAgGA8sumigSPxUvRSnzpDhBRaZH0/39pMFcY+XDjM0tVGj2ku/fdWwKHK8nwTEjNYVWGEjnFuJt9mGnacWOnyeEG9AAMvXYKrLkbKVWh+i4332F0YGqSmtzBn+2X9EQ753fujXd+s3jKkoKaYI58hb+A7BVPeVInDJOfDMmj6KklT6JFHM=
```

The appveyor.yml file has not changed since it's last successful commit.  I'll get back to that. In meantime, reset the travis file and go from there...

**Comment by timtrice** [comment]
Resolved with commit 449bee6:

- Rewrite .travis.yml and appveyor.yml with guidance from ggplot2 and
  dplyr. The primary issue was the package `sf` which had begun using
	GDAL >= 2.0. This required instructing Travis to add the
	ubuntugis-unstable repository. (ac01f31)

- The instructions on installing `rrricanesdata` were incorrect.
  Previous tests were using the `install_github` function but the
	correct call requires install.packages using the drat remote.
	(5a809d8)

- Completely removed OSX tests in Travis. Only dplyr does a travis test
  in osx and that is only one test; ggplot does zero. (ec176be)

- Added `curl` package to Imports (6027162)

- In Description, specify minimum R version as 3.3.0; a declaration of
  3.3.3 resulted in a "dependence on R not with patchlevel 0' error
	(a3a1410)

- Modify `get_nhc_link` which returned a "http" protocol; the NHC
  modified some pages (but not all) to use "https" which caused many
	tests to fail. Added parameter "protocol" to `get_nhc_link` and
	updated direct references to URLs within code to use the function
	and appropriate protocol. At this point, it seems GIS and FTP data
	is using unsecure protocol. Documentation was not updated as "http"
	will redirect to "https". (15105c7)

- Removed delays in accessing annual archives. This delay was
  unnecessary as the requested limit from the NHC is 80 requests per
	10 seconds; this would mean a delay would only be required if there
	were 80 years of data. (e7b2640)

- Added functionality to scrape the NHC's FTP servers for additional
  data. This is very beta and only gives 1997 for now (it seems).
	This will be expanded on the next minor release. No documentation
	has been updated (other than function docs).(4adb602, cdeff7d)
